pkg/operator/controller/externaldns/pod.go

```diff
@@ -544,6 +552,33 @@ func (b *externalDNSVolumeBuilder) providerSpecificVolumes() []corev1.Volume {
 	return nil
 }
 
+// awsVolumes returns volumes needed for AWS provider
+func (b *externalDNSVolumeBuilder) awsVolumes() []corev1.Volume {
+	if len(b.secretName) == 0 {
+		return nil
+	}
+	return []corev1.Volume{
+		{
+			Name: awsCredentialsVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: b.secretName,
+					Items: []corev1.KeyToPath{
```

Comment:
The items should not be present. The point of using the shared credentials is that the secret could contain static credentials or configuration for STS.

Reply:
Removed